src/core/AutomationPattern.cpp

```diff
@@ -304,7 +313,7 @@ MidiTime AutomationPattern::setDragValue( const MidiTime & time,
 	//Restore to the state before it the point were being dragged
 	m_timeMap = m_oldTimeMap;
 
-	for( timeMap::const_iterator it = m_timeMap.begin(); it != m_timeMap.end(); ++it )
+	for( timeMap::iterator it = m_timeMap.begin(); it != m_timeMap.end(); ++it )
```

Comment:
Why was this changed?

Reply:
Should the tangent be a part of the node or should it be the progression type? The tangent will be generated from the value and the progression type of the node itself. I'm not sure we need to save the tangent in the node do we?

Reply:
Currently the tangents are saved in the `m_tangents` timeMap even when the selected progression type isn't the Cubic Hermit. I guess it was done that way so the tangents don't have to be recalculated every time the pattern is being drawn, so they are stored and just updated when needed. I kept that logic but moving the tangent values to the node, since they are part of what describes the node (guess that would even open the possibility of manually editing the tangents in the future).

Reply:
It will modify `m_timeMap`, `it` is an iterator pointing to a particular member of `m_timeMap`. The `generateTangents` method receives that iterator as a parameter. Previously it was used to retrieve the values used for the tangent calculations, now it's used for both retrieving the values and for setting the tangent values. I think I could modify it to receive a QMap key instead and generate the tangent of that particular node, but I'm not sure there are any advantages to it.